### PR TITLE
feat(dashboards): Show an alert on release widgets without backfill

### PR DESCRIPTION
**Pull request by shruthilayaj**
Since we're EAin without the 90 day backfill, we want to
show an alert if users try to query for data before that
date.

Open to better error messages cc: @jas-kas 

Screen Shot 2022-06-06 at 4 48 09 PM

### CODE REVIEW

**Comment by jas-kas** [comment]
Thanks @shruthilayaj for the mention! I have a quick question: Will this alert be triggered each time a user either builds a new RH data widget **and** goes into the widget builder edit mode for a RH data widget?

Here's my copy suggestion:

> Releases data is only available from April 12, 2022. Data may be incomplete as a result.

Rationale:
- Opted for "Releases" data to match data set name
- Went with the date formatting April 12, 2022 since that's how I've seen we approach dates in Sentry ([example](https://sentry.io/organizations/sentry/alerts/rules/?expand=latestIncident&expand=lastTriggered&project=11276&sort=incident_status&sort=date_triggered&team=myteams&team=unassigned)) . It also helps avoid confusion between which is the month vs. day

**Review by jas-kas — COMMENTED** [review]
Thanks @shruthilayaj for the mention! I have a quick question: Will this alert be triggered each time a user either builds a new RH data widget and goes into the widget builder edit mode for a RH data widget?

Here's my copy suggestion:

> Releases data is only available from April 12, 2022. Data may be incomplete as a result.


Rationale:

- Opted for "Releases" data to match data set name
- Went with the date formatting April 12, 2022 since that's how I've seen we approach dates in Sentry ([example](https://sentry.io/organizations/sentry/alerts/rules/?expand=latestIncident&expand=lastTriggered&project=11276&sort=incident_status&sort=date_triggered&team=myteams&team=unassigned)). It also helps avoid confusion between which is the month vs. day

File: static/app/views/dashboardsV2/widgetCard/index.tsx [hunk 1]
```diff
@@ -212,6 +215,23 @@ class WidgetCard extends Component<Props, State> {
...
+    if (widget.widgetType === WidgetType.RELEASE && showStoredAlert) {
+      if (start) {
+        let startDate: Date | undefined = undefined;
+        if (typeof start === 'string') {
+          startDate = new Date(start);
+        } else {
+          startDate = start;
+        }
+        showIncompleteDataAlert = startDate < METRICS_BACKED_SESSIONS_START_DATE;
+      } else if (period && period === '90d') {
```
[the remark this answers is not in the document]
Comment (shruthilayaj) [reply static/app/views/dashboardsV2/widgetCard/index.tsx:229]: hmm yeah not through the page filter but they could modify the stats period on the url, will capture that edge case

**Comment by shruthilayaj** [comment]
> Thanks @shruthilayaj for the mention! I have a quick question: Will this alert be triggered each time a user either builds a new RH data widget and goes into the widget builder edit mode for a RH data widget?

yup, when they are in the widget builder and select a RH widget and it's outside the date is when the alert shows up. and will incorporate the feedback for the date!

**Review by NisanthanNanthakumar — APPROVED** [review]

**Pull request merged**